pkg/sippyserver/analyzer.go

```diff
@@ -1,122 +1,103 @@
 package sippyserver
 
 import (
-	"bytes"
-	"encoding/json"
 	"fmt"
-	"io/ioutil"
 	"regexp"
-	"strings"
 	"time"
 
-	sippyprocessingv1 "github.com/openshift/sippy/pkg/apis/sippyprocessing/v1"
 	testgridv1 "github.com/openshift/sippy/pkg/apis/testgrid/v1"
+
+	"github.com/openshift/sippy/pkg/testgridanalysis/testgridhelpers"
+
+	sippyprocessingv1 "github.com/openshift/sippy/pkg/apis/sippyprocessing/v1"
 	"github.com/openshift/sippy/pkg/buganalysis"
 	"github.com/openshift/sippy/pkg/testgridanalysis/testgridanalysisapi"
 	"github.com/openshift/sippy/pkg/testgridanalysis/testgridconversion"
-	"github.com/openshift/sippy/pkg/testgridanalysis/testgridhelpers"
 	"github.com/openshift/sippy/pkg/testgridanalysis/testreportconversion"
-	"github.com/openshift/sippy/pkg/util"
 	"github.com/openshift/sippy/pkg/util/sets"
 	"k8s.io/klog"
 )
 
-// TODO I think many of these are dead/inappropriate
-type Options struct {
-	LocalData               string
-	Releases                []string
-	StartDay                int
-	EndDay                  int
-	TestSuccessThreshold    float64
-	JobFilter               string
-	MinTestRuns             int
-	Output                  string
-	FailureClusterThreshold int
-	FetchData               string
-	ListenAddr              string
-	Server                  bool
+// TestGridLoadingOptions control the data which is loaded from disk into the testgrid structs
+type TestGridLoadingConfig struct {
+	// LocalData is the directory where the testgrid data is stored
+	LocalData string
+	// JobFilter is a regex run against job names. Only match names are loaded.
+	JobFilter *regexp.Regexp
 }
 
-type Analyzer struct {
-	Options        Options
-	LastUpdateTime time.Time
-	Release        string
-
-	BugCache buganalysis.BugCache
+// RawJobResultsAnalysisOptions control which subset of data from the testgrid data is analyzed into the rawJobResults
+type RawJobResultsAnalysisConfig struct {
+	StartDay int
+	EndDay   int
 }
 
-func (a *Analyzer) getTestGridData(releases []string, storagePath string) []testgridv1.JobDetails {
```

Comment:
this moved to https://github.com/openshift/sippy/pull/101/files#diff-9ae0ddd19e4b07eb469a96212c8537c8R83 as a straight move with args passed.